src/objectExplorer/objectExplorerProvider.ts

```diff
@@ -29,29 +31,36 @@ export class ObjectExplorerProvider implements vscode.TreeDataProvider<any> {
         return node;
     }
 
-    getChildren(element?: TreeNodeInfo): Promise<vscode.TreeItem[]> {
-        const children = this._objectExplorerService.getChildren(element);
+    async getChildren(element?: TreeNodeInfo): Promise<vscode.TreeItem[]> {
+        const children = await this._objectExplorerService.getChildren(element);
         if (children) {
-            return Promise.resolve(children);
+            return children;
         }
     }
 
-    async createSession(): Promise<string> {
-        return await this._objectExplorerService.createSession();
+    async createSession(promise: Deferred<TreeNodeInfo>, connectionCredentials?: IConnectionCredentials): Promise<void> {
+        return this._objectExplorerService.createSession(promise, connectionCredentials);
     }
```

Comment:
Just await the call here instead of returning so we're consistent with async/await style